apisix/runner/server/handle.py

```diff
@@ -0,0 +1,137 @@
+        :param debug:
+            enable debug mode
+        """
+        self.type = ty
+        self.buffer = buf
+        self.debug = debug
+
+    @property
+    def type(self) -> int:
+        return self._type
+
+    @type.setter
+    def type(self, ty: int = 0) -> None:
+        self._type = ty
+
+    @property
+    def buffer(self) -> bytes:
+        return self._buffer
+
+    @buffer.setter
+    def buffer(self, buf: bytes = b'') -> None:
+        self._buffer = buf
+
+    @property
+    def debug(self) -> bool:
+        return self._debug
+
+    @debug.setter
+    def debug(self, debug: bool = False) -> None:
+        self._debug = debug
+
+    def _rpc_config(self) -> NewServerResponse:
+        # init request
+        req = NewHttpRequest(RPC_PREPARE_CONF, self.buffer)
+        # generate token
+        token = RunnerCache.generate_token()
+        # get plugins config
+        configs = req.configs
+        # cache plugins config
+        ok = RunnerCache.set_config_by_token(token, configs)
+        if not ok:
+            return NewServerResponse(code=A6ErrCode.Code.SERVICE_UNAVAILABLE, message="cache token failure")
+        # init response
+        resp = NewHttpResponse(RPC_PREPARE_CONF)
+        resp.token = token
+        response = resp.flatbuffers()
+
+        return NewServerResponse(code=RUNNER_SUCCESS_CODE, message=RUNNER_SUCCESS_MESSAGE, data=response.Output(),
+                                 ty=self.type)
+
+    def _rpc_call(self) -> NewServerResponse:
+        # init request
+        req = NewHttpRequest(RPC_HTTP_REQ_CALL, self.buffer)
+        # get request token
+        token = req.conf_token
+        # get plugins
+        configs = RunnerCache.get_config_by_token(token)
+        if len(configs) == 0:
+            return NewServerResponse(code=A6ErrCode.Code.CONF_TOKEN_NOT_FOUND, message="cache token not found")
+        # init response
+        resp = NewHttpResponse(RPC_HTTP_REQ_CALL)
+        # execute plugins
+        RunnerPlugin.filter(configs, req, resp)
+
+        response = resp.flatbuffers()
+        return NewServerResponse(code=RUNNER_SUCCESS_CODE, message=RUNNER_SUCCESS_MESSAGE, data=response.Output(),
+                                 ty=self.type)
+
+    @staticmethod
+    def _rpc_unknown(err_code: int = 0) -> NewServerResponse:
+        resp = NewHttpResponse(RPC_UNKNOWN)
+        resp.error_code = err_code
+        response = resp.flatbuffers()
+        return NewServerResponse(code=RUNNER_SUCCESS_CODE, message="OK", data=response.Output(),
+                                 ty=RPC_UNKNOWN)
+
+    def dispatch(self) -> NewServerResponse:
+        err = None
+
+        if self.type == RPC_PREPARE_CONF:
+            err = self._rpc_config()
+
+        if self.type == RPC_HTTP_REQ_CALL:
+            err = self._rpc_call()
+
+        if not err:
+            return self._rpc_unknown()
+
+        size = len(err.data)
+        if (size > RESP_MAX_DATA_SIZE or size <= 0) and err.code == 200:
```

Comment:
Why err.code != 200 can bypass the max size check?

Reply:
If an exception is returned, the processing error will be responded to first, and then the size will be checked. the size check will not be skipped.